File: src/main/java/data/model/FootballerDaoImpl.java
package data.model;


import com.squareup.moshi.JsonAdapter;
import com.squareup.moshi.Moshi;
import org.json.JSONArray;

import java.io.IOException;
import java.util.ArrayList;
import java.util.List;
import java.util.Set;
import java.util.stream.Collectors;

public class FootballerDaoImpl implements FootballerDao {

    //List<Footballer> footballers = new ArrayList<>();

    public List<Footballer> getAll(int managerId, int gameWeek) throws IOException {
        JSONArray picks = new PicksDao().getPicks(managerId, gameWeek - 1);
        Moshi moshi = new Moshi.Builder().build();
        JsonAdapter<Pick> picksAdapter = moshi.adapter(Pick.class);

        List<Footballer> footballers = new ArrayList<>();
        for (int i = 0; i < picks.length(); i++) {
            Pick pick = picksAdapter.fromJson((picks.get(i)).toString());
            if (pick != null) {
                Footballer footballer = new Footballer();
                footballer.setId(pick.element);
                footballer.setPosition(pick.position);
                footballers.add(footballer);
            }
        }
        return footballers;
    }

    public void update(List<Footballer> footballers, int id, Opponent opponent) {
        Footballer footballer = footballers.stream()
                .filter(x -> id == x.getId())
                .limit(1)
                .collect(Collectors.toList())
                .get(0);
        footballer.getOpponentList().add(opponent);
    }

    public List<Footballer> getInverse(List<Footballer> footballers, List<Footballer> toInvert) {
        Set<Integer> existingIds = toInvert
                .stream()
                .map(Footballer::getId)
                .collect(Collectors.toSet());

        return footballers
                .stream()
                .filter(footballer -> !existingIds.contains(footballer.getId()))
                .collect(Collectors.toList());
    }


    public List<Footballer> getByTeamId(List<Footballer> footballers, int teamId) {
        return footballers.stream()
                .filter(x -> teamId == x.getTeamId())
                .limit(3)
                .collect(Collectors.toList());
    }

}

File: src/main/java/data/model/FootballerDaoImpl.kt
package data.model


import com.squareup.moshi.Moshi
import java.io.IOException
import java.util.*
import java.util.stream.Collectors

class FootballerDaoImpl : FootballerDao {

    @Throws(IOException::class)
    override fun getAll(managerId: Int, gameWeek: Int): MutableList<Footballer> {
        val pickAdapter = Moshi.Builder().build().adapter(Pick::class.java)
        val pickArray = PickDao().getPicks(managerId, gameWeek)
        val footballers = ArrayList<Footballer>()
        for (i in 0 until pickArray.length()) {
            val pick = pickAdapter.fromJson(pickArray.get(i).toString())
            if (pick != null) {
                val footballer = Footballer()
                footballer.id = pick.element
                footballer.position = pick.position
                footballers.add(footballer)
            }
        }
        return footballers
    }

    override fun update(footballers: List<Footballer>, id: Int, opponent: Opponent) {
        val footballer = footballers.stream()
                .filter { x -> id == x.id }
                .limit(1)
                .collect(Collectors.toList())[0]
        footballer.opponentList.add(opponent)
    }

    fun getInverse(footballers: List<Footballer>, toInvert: List<Footballer>): List<Footballer> {
        val existingIds = toInvert
                .stream()
                .map(Footballer::id)
                .collect(Collectors.toSet())

        return footballers
                .stream()
                .filter { footballer -> !existingIds.contains(footballer.id) }
                .collect(Collectors.toList())
    }

    fun getByTeamId(footballers: List<Footballer>, teamId: Int): List<Footballer> {
        return footballers.stream()
                .filter { x -> teamId == x.teamId }
                .limit(3)
                .collect(Collectors.toList())
    }

}
